dashboard/app/controllers/pd/misc_survey_controller.rb

```diff
@@ -0,0 +1,77 @@
+module Pd
+  class MiscSurveyController < ApplicationController
+    include JotForm::EmbedHelper
+
+    # The POST submit route will be redirected to from JotForm, after form submission
+    skip_before_action :verify_authenticity_token, only: %w(submit)
+
+    # GET /pd/misc_survey/thanks
+    def thanks
+    end
+
+    # General misc survey.
+    # GET '/pd/misc/:form_id'
+    #
+    # The JotForm survey will redirect to thanks.
+    def new
+      @form_id = params[:form_id]
```

Comment:
What do you see if the form_id is invalid?

Reply:
I have modified the system to now use a tag rather than a form ID in the URL, and it will render a 404 if the tag is not one that is explicitly allowed to be embedded.